C++/test/algorithm/dynamic_programming/longest_subsequence.cpp

```diff
@@ -0,0 +1,15 @@
+#include "third_party/catch.hpp"
+#include "algorithms/dynamic_programming/longest_subsequence.hpp"
+
+using namespace std;
+
+TEST_CASE("Base cases", "[longest_subsequence]"){
+	REQUIRE(longest_subsequence(vector<int>({0})) == 1);
+	REQUIRE(longest_subsequence(vector<int>({})) == 0);
+}
+
+TEST_CASE("non Base cases", "[longest_subsequence]"){
```

Comment:
How about test cases for increasing subsequences and decreasing subsequences? Should also probably have some for a list of integers which are neither increasing nor decreasing (all the same).